sale_automatic_workflow/models/automatic_workflow_job.py

```diff
@@ -54,6 +54,27 @@ def _do_validate_sale_order(self, sale, domain_filter):
         sale.action_confirm()
         return "{} {} confirmed successfully".format(sale.display_name, sale)
 
+    def _do_send_order_confirmation_mail(self, sale):
+        """Send order confirmation mail, while filtering to avoid duplication if the
+        message was already sent in in action_confirm() function from sale.order model
+        in website_sale"""
+        if not self.env["sale.order"].search_count(
+            [("id", "=", sale.id), ("state", "=", "sale")]
+        ):
+            return "{} {} job bypassed".format(sale.display_name, sale)
+        if (
```

Comment:
Why do you need this extra check?

Reply:
You can probably reduce this condition a bit, but it is necessary to maintain it in order to avoid that an email is sent 2 times in the event that the order has been self-confirmed because the transaction has been done.

Thinking about it carefully, this function only executes when trying to validate an order according to a specified domain; Therefore, if you have reached this point, the email should always be sent because if it were a self-confirmed e-commerce order (transaction done) it would not have reached here so this check could probably be removed.

What do you think about it?

Reply:
You tell me then...

Reply:
Now it's ready.